aspnetcore/signalr/configuration.md

```diff
@@ -0,0 +1,234 @@
+    .build();
+```
+
+### Configure bearer authentication
+
+To provide authentication data along with SignalR requests, use the `AccessTokenProvider` option (`accessTokenFactory` in JavaScript) to specify a function that returns the desired access token. In the .NET Client, this access token is passed in as an HTTP "Bearer Authentication" token (Using the `Authorization` header with a type of `Bearer`). In the JavaScript client, the access token is used as a Bearer token, **except** in a few cases where browser APIs restrict the ability to apply headers (specifically, in Server-Sent Events and WebSockets requests). In these cases, the access token is provided as a query string value `access_token`.
+
+In the .NET client, the `AccessTokenProvider` option can be specified using the options delegate in `WithUrl`:
+
+```csharp
+var connection = new HubConnectionBuilder()
+    .WithUrl("https://example.com/myhub", options => {
+        options.AccessTokenProvider = async () => {
+            // Get and return the access token.
+        }
+    })
+    .Build();
+```
+
+In the JavaScript client, the access token is configured by setting the `accessTokenFactory` field on the options object in `withUrl`:
+
+```javascript
+let connection = new signalR.HubConnectionBuilder()
+    .withUrl("/myhub", {
+        accessTokenFactory: () => {
+            // Get and return the access token.
+            // This function can return a JavaScript Promise if asynchronous
+            // logic is required to retrieve the access token.
+        }
+    })
+    .build();
+```
+
+### Configure timeout and keep-alive options
+
+Additional options for configuring timeout and keep-alive behavior are available on the `HubConnection` object itself:
+
+| .NET Option | JavaScript Option | Description |
+| ----------- | ----------------- | ----------- |
+| `ServerTimeout` | `serverTimeoutInMilliseconds` | Timeout for server activity. If the server hasn't sent any message in this interval, the client considers the server disconnected and trigger the `Closed` event (`onclose` in JavaScript). |
+| `HandshakeTimeout` | Not configurable | Timeout for initial server handshake. If the server doesn't send a handshake response in this interval, the client cancels the handshake and trigger the `Closed` event (`onclose` in JavaScript). |
+
+In the .NET Client, timeout values are specified as `TimeSpan` values. In the JavaScript client, timeout values are specified as numbers. The numbers represent time values in milliseconds.
+
+### Configure additional options
+
+Additional options can be configured in the `WithUrl` (`withUrl` in JavaScript) method on `HubConnectionBuilder`:
+
+| .NET Option | JavaScript Option | Description |
+| ----------- | ----------------- | ----------- |
+| `AccessTokenProvider` | `accessTokenFactory` | A function returning a string that is provided as a Bearer authentication token in HTTP requests. |
```

Comment:
There are some options missing:
`ClientCertificates`
`CloseTimeout`
`HttpMessageHandlerFactory`
`Transports`
`Url`
`UseDefaultCredentials`

Reply:
For `Url` and `Transports`, I didn't want to confuse users by the fact that there are multiple ways to configure those options. The overloads allow them to be set just fine.